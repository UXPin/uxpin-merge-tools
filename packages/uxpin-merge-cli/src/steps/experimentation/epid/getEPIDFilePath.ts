import { resolve } from 'path';
import { TEMP_DIR_PATH } from '../../building/config/getConfig';

export const EPID_FILENAME = 'epid';

export function getEPIDFilePath(projectRoot: string): string {
  return resolve(projectRoot, TEMP_DIR_PATH, EPID_FILENAME);
}
