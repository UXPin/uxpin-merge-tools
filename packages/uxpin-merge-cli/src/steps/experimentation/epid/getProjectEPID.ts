import { readFile } from 'fs-extra';
import { EPID } from './EPID';

export async function getProjectEPID(filePath: string): Promise<EPID> {
  const content: string = await readFile(filePath, { encoding: 'utf-8' });
  return JSON.parse(content);
}
