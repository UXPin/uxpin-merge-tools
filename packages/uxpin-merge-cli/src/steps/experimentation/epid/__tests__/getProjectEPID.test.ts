import * as path from 'path';
import { getEPIDFilePath } from '../getEPIDFilePath';
import { getProjectEPID } from '../getProjectEPID';

describe('getProjectEPID', () => {
  it('should read epid file and convert it to EPID object', async () => {
    // given
    const projectPath: string = path.resolve(__dirname, '../../../../../test/resources/designSystems/withEpidFile/');
    const filePath: string = getEPIDFilePath(projectPath);

    // when
    // then
    expect(await getProjectEPID(filePath)).toMatchSnapshot();
  });

  it("should throws error when epid file doesn't exists", async () => {
    // given
    const projectPath = './not/exists';
    const filePath: string = getEPIDFilePath(projectPath);

    // when
    // then
    await expect(getProjectEPID(filePath)).rejects.toThrowError(/no such file or directory/);
  });
});
