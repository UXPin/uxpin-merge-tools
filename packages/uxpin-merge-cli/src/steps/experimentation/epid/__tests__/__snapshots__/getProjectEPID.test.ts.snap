// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProjectEPID should read epid file and convert it to EPID object 1`] = `
{
  "revisionId": "1c8e56d6-e4a4-4a16-8651-015124f10c83_b2c0fce45b547aa926428eab6a29ba66ea4ca178",
}
`;
